src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/CSharpPullMemberUpCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PullMemberUp
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(PredefinedCodeRefactoringProviderNames.PullMember)), Shared]
+    internal class CSharpPullMemberUpCodeRefactoringProvider : AbstractPullMemberUpRefactoringProvider
+    {
+        /// <summary>
+        /// For test purpose only.
+        /// </summary>
+        public CSharpPullMemberUpCodeRefactoringProvider(IPullMemberUpOptionsService pullMemberUpService) : base(pullMemberUpService)
+        {
+        }
+
+        internal CSharpPullMemberUpCodeRefactoringProvider() : this(null)
+        {
+        }
+
+        internal override bool IsUserSelectIdentifer(SyntaxNode userSelectedSyntax, CodeRefactoringContext context)
```

Comment:
1. gramatically, this sounds odd.  Perhaps: IsSelectionValid?  
2. you don't really need/use the full 'context' object.  instead just pass the span in.  it will help with the name IsSelectionValid as well.
3. in general, try to preserve 'declared order' of values.  So, in the caller, they had the 'context' object decalred before getting the syntax.  So this should pass in the context obj first, then the syntax node.

Reply:
Make sense.  But there are so many methods so I will try to 'sort' the order of the value gradually.